Authentication/Login/View/LoginController.swift

```diff
@@ -174,35 +173,35 @@ extension LoginController: UITextFieldDelegate {
         _activeField.value = textField
     }
     
-    public func textFieldDidEndEditing(textField: UITextField) {
+    public func textFieldDidEndEditing(_ textField: UITextField) {
         if textField == loginView.emailTextField {
             loginView.emailTextFieldSelected = false
         } else {
             loginView.passwordTextFieldSelected = false
         }
-        _activeField.value = .None
+        _activeField.value = .none
     }
     
 }
 
 private extension LoginController {
     
-    private func addKeyboardObservers() {
+    func addKeyboardObservers() {
         _disposable += _keyboardDisplayed <~ _notificationCenter
-            .rac_notifications(UIKeyboardDidHideNotification)
+            .reactive.notifications(forName: NSNotification.Name.UIKeyboardDidHide)
             .map { _ in false }
         
         _disposable += _notificationCenter
-            .rac_notifications(UIKeyboardWillShowNotification)
-            .startWithNext { [unowned self] in self.keyboardWillShow($0) }
+            .reactive.notifications(forName: NSNotification.Name.UIKeyboardWillShow)
+            .observeValues { [unowned self] in self.keyboardWillShow($0) }
         
         _disposable += _notificationCenter
-            .rac_notifications(UIKeyboardWillHideNotification)
-            .startWithNext { [unowned self] _ in self.view.frame.origin.y = 0 }
+            .reactive.notifications(forName: NSNotification.Name.UIKeyboardWillHide)
+            .observeValues { [unowned self] _ in self.view.frame.origin.y = 0 }
     }
     
-    private func keyboardWillShow(notification: NSNotification) {
-        if let keyboardSize = (notification.userInfo?[UIKeyboardFrameEndUserInfoKey] as? NSValue)?.CGRectValue() {
+    func keyboardWillShow(_ notification: Notification) {
```

Comment:
Without it, it would use `notification:` as the label, but it reads kinda weirdly: `keyboardWillShow(notification:)`: sounds like the keyboard will show a notification. I'd leave it like this, but we can discuss it further if you don't like it.